### ISSUE TITLE
RuleTester detection seems like it could use improving

### ISSUE THREAD

**Issue by G-Rath** [not-an-aardvark/eslint-plugin-eslint-plugin#248]
I've noticed the `RuleTester` detection logic seems to trip up on what I'd consider common or easy to support cases - the latest was that it didn't flag [this test](https://github.com/jest-community/eslint-plugin-jest/pull/1094/commits/a12fb7e745b7f0af5f70e6f7a05370981a488ce0) as duplicated because I've got the tests wrapped in a `describe`.

e.g.

```
describe('reference tests', () => {
  ruleTester.run('esm', rule, {
    valid: ['not flagged', 'not flagged'],
    invalid: [],
  });
});

ruleTester.run('esm', rule, {
  valid: ['not flagged', 'not flagged'],
  invalid: [],
});
```

I think there was at least another instance where it didn't seem to pick up alternative names, but it's been a while so can't remember the details.

I've not yet had the time to dig into this that deeply but wanted to open this issue in the meantime to get some discussion going - I'm happy to help with trying to improve the logic.

**Comment by bmish** [comment]
To be clear, are you talking about the [no-identical-tests](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/no-identical-tests.md) rule?

**Comment by G-Rath** [comment]
That and at least the test case properties rules as they're also impacted in the same way. I've not had the time yet to do any real testing to check how many rules it affects, and to come up with some common cases I think it would be nice to try and support (it might turn out that this is the main one that's easy to do 🤷)

**Comment by aladdin-add** [comment]
yes, all the tests-related rules were behaving the same way - it is only checking the top-level `ruleTester.run()`.  https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/b685b2aa9bb6c88d752dde7a9ddaf4c03efbc226/lib/utils.js#L389

good to check a few more common cases. 👍
